.docs/content/1.concepts/3.execute-tests.md

```diff
@@ -0,0 +1,119 @@
+
+# How to execute tests in ArmoniK Core?
+
+## Unit tests independent of deployment
+
+For this type of tests, you can directly execute it. The first option is to run the tests using your code editor. The second is to open a terminal, go to the folder where the test project is located and do:
+
+```shell
+dotnet test
+```
+
+## Unit tests that need partial deployment
+
+First, you should do the needed partial deployment that deploys the container running needed service.
+For example, for the case of ArmoniK.Core.Adapters.RabbitMQ.Tests, execute:
+
+```shell
+just queue=rabbitmq deployTargetQueue
+```
+
+For ArmoniK.Core.Adapters.LocalStorage.Tests deploy with local object storage:
+
+```shell
+just object=local deployTargetObject
+```
+
+And so you can execute the corresponding tests:
+
+- via your code editor
+- or in your command line, go to the folder where the test project is located and do:
+
+```shell
+dotnet test
+```
+
+## Integration tests
+
+HtcMock, Stream and Bench are integration tests that need a Core deployment. To be able to run these tests, you need first to deploy ArmoniK Core with the appropriate values for the parameter `worker`: htcmock, stream or bench. If you don't specify any value for this parameter, it's set to htcmock by default. Don't forget to use the recipe to build the images for your deployments if they do not exist.
+
+### Htcmock
+
+The default deployment is done with htcmock worker:
+
+```shell
+just deploy
+```
+
+After this deployment, there is a `just` recipe that allows to build the docker image to test htcmock worker, you can use this recipe to build the image:
+
+```shell
+just buildHtcmockClient
+```
+
+Finally execute the test using:
+
+```shell
+docker run --net armonik_network --rm \
+            -e HtcMock__NTasks=100 \
+            -e HtcMock__TotalCalculationTime=00:00:00.100 \
+            -e HtcMock__DataSize=1 \
+            -e HtcMock__MemorySize=1 \
+      	    -e HtcMock__EnableFastCompute=true \
+            -e HtcMock__SubTasksLevels=1 \
+            -e HtcMock__Partition=TestPartition0 \
+            -e GrpcClient__Endpoint=http://armonik.control.submitter:1080 \
+             dockerhubaneo/armonik_core_htcmock_test_client:TAG
+
+```
+You will need to replace `TAG` by the tag of your image `armonik_core_htcmock_test_client`. If you don't specify any tag for `just buildHtcmockClient`, the tag will be `test`, but you are given the option to use your own tag:
+
+```shell
+just tag=mytag buildHtcmockClient
+```
+
+### Stream
+
+Deploy using stream worker:
+
+```shell
+just worker=stream deploy
+```
+
+Use the recipe to build the docker image for stream worker:
+
+```shell
+just buildStreamClient
+```
+
+Now, you can execute the tests using your code editor, or via the command line. In the latter case, go to the folder where the tests are located and type:
```

Comment:
It will nice to add an example of a simple test script for stream like you did for htcmock and bench. Something like that:
docker run --rm  --net armonik_network --rm \
             -e GrpcClient__Endpoint="${CONTROL_PLANE_URL}" \
             -e Partition=TestPartition0 \
             -e GrpcClient__Endpoint=http://armonik.control.submitter:1080 \
             dockerhubaneo/armonik_core_stream_test_client:0.12.0